### PR TITLE
Add Username + Password Authentication for Trino Client (fixes #829)

### LINKED ISSUE #829: Add basic authorization into trino client
Probably trino client should use username/password authorization, like presto client do

### PR DESCRIPTION
**Pull request by rbb17**
Adding username and password authentication for the trino client (fixes https://github.com/pinterest/querybook/issues/829)

Like the presto client-
[querybook/querybook/server/lib/query_executor/clients/presto.py](https://github.com/pinterest/querybook/blob/1e3d01e8a8177f7c4a0031b681f4e1e8d95b3f76/querybook/server/lib/query_executor/clients/presto.py#L12)

### CODE REVIEW

File: querybook/server/lib/query_executor/clients/trino.py [hunk 1]
```diff
@@ -61,12 +62,15 @@ def __init__(
         host = trino_conf.host
         port = 8080 if not trino_conf.port else trino_conf.port
 
+        auth = trino.auth.BasicAuthentication(username, password)
```
Comment (czgu) [inline querybook/server/lib/query_executor/clients/trino.py:65]: to make the behavior consistent as before, can you only provide `auth` to connect params if both username and password are provided?

Comment (rbb17) [reply querybook/server/lib/query_executor/clients/trino.py:65]: Ack

File: querybook/server/lib/query_executor/clients/trino.py [hunk 2]
```diff
@@ -61,14 +62,27 @@ def __init__(
...
+            )
+        else:
+            connection = trino.dbapi.connect(
+                host=host,
+                port=port,
+                catalog=trino_conf.catalog,
+                schema=trino_conf.schema,
+                user=proxy_user or username,
+                http_scheme=trino_conf.protocol,
+            )
```
Comment (czgu) [inline querybook/server/lib/query_executor/clients/trino.py:85]: no need to duplicate this code
```suggestion
        auth = trino.constants.DEFAULT_AUTH
        if username is not None and password is not None:
            auth = trino.auth.BasicAuthentication(username, password)

        connection = trino.dbapi.connect(
            host=host,
            port=port,
            catalog=trino_conf.catalog,
            schema=trino_conf.schema,
            auth=auth,
            user=proxy_user if proxy_user else username,
            http_scheme=trino_conf.protocol,
        )
```

Comment (rbb17) [reply querybook/server/lib/query_executor/clients/trino.py]: Done

**Review by czgu — APPROVED** [review]

**Comment by rbb17** [comment]
@czgu Can you merge this pls. // @Aka-shi

**Pull request merged**